File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        count = 0
        
        def column(j):
            col =[i[j] for i in grid]
            return col
        
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i] == column(j):
                    count += 1
                    
        return count
            

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.cpp
class Solution {
public:
    int equalPairs(vector<vector<int>>& grid) {
        int n = grid.size();
        int answer = 0;
        map<vector<int>, int> rows;
        
        for(int i = 0; i < n; i++)
        {
            vector<int> row(n);
            
            for(int j = 0; j < n; j++)
                row[j] = grid[i][j];
            
            rows[row]++;
        }
        
        for(int j = 0; j < n; j++)
        {
            vector<int> col(n);
            
            for(int i = 0; i < n; i++)
                col[i] = grid[i][j];
            
            if(rows.find(col) != rows.end())
                answer += rows[col];
        }
        
        return answer;
    }
};